apps/wallet/src/ui/app/components/known-scam-overlay/useDappPreflight.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { useAppsBackend } from '@mysten/core';
+import { useSuiClient } from '@mysten/dapp-kit';
+import { type Transaction } from '@mysten/sui/transactions';
+import { toB64 } from '@mysten/sui/utils';
+import { useQuery } from '@tanstack/react-query';
+
+import { RequestType, type DappPreflightRequest, type DappPreflightResponse } from './types';
+
+export function useDappPreflight({
+	requestType,
+	origin,
+	transaction,
+	message,
+}: {
+	requestType: RequestType;
+	origin?: string;
+	transaction?: Transaction;
+	message?: string;
+}) {
+	const { request } = useAppsBackend();
+	const client = useSuiClient();
+
+	return useQuery({
+		queryKey: ['dapp-preflight', { requestType, origin, transaction, message }],
```

Comment:
Transaction isn’t able to be safely serialized like this so you can’t use it as a cache key directly

Reply:
Updated to use the `requestId` as a key which is unique per approval request